waspc/data/Generator/templates/server/src/auth/session.ts

```diff
@@ -0,0 +1,102 @@
+{{={= =}=}}
+import { Request as ExpressRequest } from "express";
+
+import { type {= userEntityUpper =} } from "../entities/index.js"
+import { type SanitizedUser } from '../_types/index.js'
+
+import { auth } from "./lucia.js";
+import {
+  throwInvalidCredentialsError,
+  deserializeAndSanitizeProviderData,
+} from "./utils.js";
+
+import prisma from '../dbClient.js';
+
+// Lucia's Session is tied to the Auth model,
+// so we keep the Auth ID in the session.
+export async function createSession(authId: string) {
+  return auth.createSession(authId, {});
+}
+
+export function getSessionAndUserFromBearerToken(req: ExpressRequest) {
+  const authorizationHeader = req.headers["authorization"];
+
+  if (typeof authorizationHeader !== "string") {
+    return {
+      user: null,
+      session: null,
+    };
+  }
+
+  const sessionId = auth.readBearerToken(authorizationHeader ?? "");
+  if (!sessionId) {
+    return {
+      user: null,
+      session: null,
+    };
+  }
+
+  return getSessionAndUserFromSessionId(sessionId);
+}
+
+export async function getSessionAndUserFromSessionId(sessionId: string) {
+  const { session, user: authEntity } = await auth.validateSession(sessionId);
+
+  if (!session || !authEntity) {
+    return {
+      user: null,
+      session: null,
+    };
+  }
+
+  return {
+    session,
+    user: await getUser(authEntity.userId)
+  }
+}
+
+async function getUser(userId: {= userEntityUpper =}['id']): Promise<SanitizedUser> {
```

Comment:
Moved from `core/auth.js` and added types. 

Now the `SanitizedUser` type and the `getUser` function are finally connected and changes in one place will cause type errors in the other place.